frontend/public/components/catalog/catalog-items.jsx

```diff
@@ -12,101 +12,151 @@ import {Modal} from 'patternfly-react/dist/js/components/Modal';
-      showAllItemsForCategory: null,
-      activeTabs,
-      filters,
-      filterCounts,
-      showOverlay: false,
-    });
-
     this.openOverlay = this.openOverlay.bind(this);
     this.closeOverlay = this.closeOverlay.bind(this);
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { filters, filterCounts, activeTabs, categories } = this.state;
+  componentDidMount () {
+    this.setState(this.getUpdatedStateFromURL(window.location.search));
+  }
+
+  componentDidUpdate(prevProps) {
+    const { filters, selectedCategory } = this.state;
     const { items } = this.props;
 
     if (items !== prevProps.items) {
-      const newCategories = categorizeItems(items);
-      this.setState(this.getCategoryState(activeTabs, newCategories));
-      if (this.hasActiveFilters(filters)) {
-        const filteredItems = this.filterItems(items);
-        const filteredCategories = recategorizeItems(filteredItems, newCategories);
-        this.setState(this.getCategoryState(activeTabs, filteredCategories));
-      }
-      this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, newCategories));
+      const categories = categorizeItems(items);
+      this.setState(this.getUpdatedState(categories, selectedCategory, filters));
     }
+  }
 
-    if (filters !== prevState.filters) {
-      const filteredItems = this.filterItems(items);
-      const newCategories = recategorizeItems(filteredItems, categories);
-      this.setState(this.getCategoryState(activeTabs, newCategories));
-    }
+  getUpdatedState (categories, selectedCategory, filters) {
+    const { items } = this.props;
+    const { filterCounts } = this.state || {};
 
-    if (activeTabs !== prevState.activeTabs) {
-      this.setState(this.getCategoryState(activeTabs, categories));
+    if (!items) {
+      return;
     }
 
-    // filter counts are updated when new Category tab is selected or filter by name changed
-    if (activeTabs !== prevState.activeTabs || prevState.filters.byKeyword !== filters.byKeyword ) {
-      this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, categories));
+    const updateFilterCounts = filterCounts ||
+      {
+        byType: {
+          clusterServiceClasses: 0,
+          imageStreams: 0,
+        },
+      };
+
+    const filteredItems = this.filterItems(items, filters);
+
+    const newCategories = recategorizeItems(filteredItems, categories);
+
+    return {
+      filters,
+      ...CatalogTileViewPage.getCategoryState(selectedCategory, newCategories),
+      filterCounts: this.getFilterCounts(selectedCategory, filters, updateFilterCounts, newCategories)
+    };
+  }
+
+  getUpdatedStateFromURL(searchURL) {
+    const { categories } = this.state;
+    const searchParams = new URLSearchParams(searchURL);
+    const categoryParam = searchParams.get(CATEGORY_URL_PARAM);
+    let keywordParam = searchParams.get(KEYWORD_URL_PARAM);
+    let byTypeParam = searchParams.get(TYPE_URL_PARAM);
```

Comment:
`byTypeParam` is modified below, cannot be const

Reply:
My suggestion is to not modify it. See the lines above ^

Reply:
Yes! It's not about `let` and `const`. The problem is you're changing the *type* of the variable from string to array, which generally should be avoided and can lead to bugs. This would be a compiler error in TypeScript, by the way.

I see these are two different values fundamentally: the query string value and then the array of type filters. They should be tracked by different variables, and things are easier to understand that way (even if we didn't have `const`).

Reply:
Ok, makes sense, fixed -thanks